### PR TITLE
Add package download from remote location

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

**Review by ctienshi — APPROVED** [review]

File: modules/apim_common/manifests/init.pp [hunk 1]
```diff
@@ -45,9 +45,20 @@
...
 
   # Copy JDK to Java distribution path
-  file { "jdk-distribution":
-    path   => "${java_home}.tar.gz",
-    source => "puppet:///modules/${module_name}/jdk/${jdk_name}.tar.gz",
+  if $pack_location == "local" {
+    file { "jdk-distribution":
+      path   => "${java_home}.tar.gz",
+      source => "puppet:///modules/${module_name}/jdk/${jdk_name}.tar.gz",
+      notify  => Exec["unpack-jdk"],
```
Comment (VimukthiPerera) [inline modules/apim_common/manifests/init.pp]: Please format the code.

File: modules/apim_common/manifests/params.pp [hunk 2]
```diff
@@ -19,6 +19,10 @@
   $packages = ["unzip"]
   $version = "2.6.0"
 
+  $pack_location = "local"
+  # $pack_location = "remote"
```
Comment (VimukthiPerera) [inline modules/apim_common/manifests/params.pp:24]: Shall we add a comment explaining what `pack_location` is about?

File: modules/apim_common/manifests/params.pp [hunk 3]
```diff
@@ -41,46 +45,54 @@
   case $profile {
     'apim_analytics_dashboard': {
       $pack = "wso2am-analytics-${version}"
+      # $remote_pack = "http://10.100.5.174:8000/packs/wso2am-analytics-2.6.0.zip"
```
Comment (VimukthiPerera) [inline modules/apim_common/manifests/params.pp]: Shall we remove the IP address and add a placeholder? WDYT?

Comment (Savidude) [reply modules/apim_common/manifests/params.pp]: Fixed https://github.com/wso2/puppet-apim/pull/128/commits/775a688e3eae96b3f6c013f85e1fed7bd19ae827

File: modules/apim_common/manifests/params.pp (hunk 2 above)
Comment (Savidude) [reply modules/apim_common/manifests/params.pp:24]: Fixed https://github.com/wso2/puppet-apim/pull/128/commits/775a688e3eae96b3f6c013f85e1fed7bd19ae827

File: modules/apim_common/manifests/init.pp (hunk 1 above)
Comment (Savidude) [reply modules/apim_common/manifests/init.pp]: Fixed https://github.com/wso2/puppet-apim/pull/128/commits/775a688e3eae96b3f6c013f85e1fed7bd19ae827

**Review by VimukthiPerera — APPROVED** [review]

**Pull request merged**